File: HackerRank/Interview_Prep/Arrays/2D_Array_DS/solution.py
#!/bin/python3

import math
import os
import random
import re
import sys

# Complete the hourglassSum function below.
def hourglassSum(a):
    n = len(a)
    maxSum = -100
    s = 0
    for i in range(n-2):
        for j in range(n-2):
            s = a[i][j] + a[i][j+1] + a[i][j+2] + a[i+1][j+1] + a[i+2][j] + a[i+2][j+1] + a[i+2][j+2]
            #print(s)
            if s > maxSum:
                maxSum = s
    return maxSum

File: HackerRank/Interview_Prep/Arrays/2D_Array_DS/solution.cpp
#include <bits/stdc++.h>

using namespace std;

// Complete the hourglassSum function below.
int hourglassSum(vector<vector<int>> arr) {
    int maxSum = -100;
    int s = 0;
    for(int i=0; i<arr.size()-2; ++i)
    {
        for(int j=0; j<arr.size()-2;++j)
        {
            s = arr[i][j] + arr[i][j+1] + arr[i][j+2] + arr[i+1][j+1] + arr[i+2][j] + arr[i+2][j+1] + arr[i+2][j+2];
            if (s > maxSum)
            {
                maxSum = s;
            }
        }
    }
    return maxSum;
}